File: backend/src/main/java/com/gxs/bank/model/UpiId.java
package com.gxs.bank.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "upi_ids")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UpiId {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private SavingsAccount account;

    @Column(name = "upi_address", nullable = false, unique = true)
    private String upiAddress;

    @Column(name = "is_primary")
    private Boolean isPrimary;

    @Column(name = "is_active")
    private Boolean isActive;

    @Column(name = "daily_limit", precision = 15, scale = 2)
    private BigDecimal dailyLimit;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (isPrimary == null) isPrimary = true;
        if (isActive == null) isActive = true;
        if (dailyLimit == null) dailyLimit = new BigDecimal("100000.00");
    }
}

File: backend-python/com/gxs/bank/model/UpiId.py
from datetime import datetime
from decimal import Decimal
import uuid

from sqlalchemy import Boolean, Column, DateTime, ForeignKey, Numeric, String
from sqlalchemy.orm import relationship

from com.gxs.bank.config.database import Base


class UpiId(Base):
    __tablename__ = "upi_ids"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    userId = Column("user_id", String(36), ForeignKey("users.id"), nullable=False)
    user = relationship("User", lazy="select")

    accountId = Column("account_id", String(36), ForeignKey("savings_accounts.id"), nullable=False)
    account = relationship("SavingsAccount", lazy="select")

    upiAddress = Column("upi_address", String(255), nullable=False, unique=True)
    isPrimary = Column("is_primary", Boolean, nullable=False, default=True)
    isActive = Column("is_active", Boolean, nullable=False, default=True)
    dailyLimit = Column("daily_limit", Numeric(15, 2), nullable=False, default=Decimal("100000.00"))
    createdAt = Column("created_at", DateTime, nullable=False, default=datetime.utcnow)
